zentral/contrib/osquery/serializers.py

```diff
@@ -220,3 +221,45 @@ def update(self, instance, validated_data):
         if compliance_check_deleted:
             query.refresh_from_db()
         return query
+
+
+class PackQuerySerializer(serializers.ModelSerializer):
+    pack = serializers.PrimaryKeyRelatedField(queryset=Pack.objects.all())
+    query = serializers.PrimaryKeyRelatedField(queryset=Query.objects.all())
+
+    class Meta:
+        model = PackQuery
+        fields = "__all__"
+
+    def validate_query(self, value):
+        slug = slugify(value.name)
+        pq_qs = PackQuery.objects.all()
+        if self.instance:
+            pq_qs = pq_qs.exclude(id=self.instance.id)
+        if pq_qs.filter(slug=slug).exists():
+            raise serializers.ValidationError(["PackQuery with this Query already exists."])
+        return value
+
+    def validate_log_removed_actions(self, value):
+        if value and self.initial_data.get("snapshot_mode"):
```

Comment:
If you need to validate values that depend on each others, you cannot use the `validate_` methods. You see this since you need to get the `initial_data` that might be changed by the data actually used by the serializer. See [the DRF docs](https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation).